File: app/src/main/java/com/chooloo/www/callmanager/ui/settings_fragment/SettingsPresenter.java
package com.chooloo.www.callmanager.ui.settings_fragment;

import androidx.preference.Preference;

import com.chooloo.www.callmanager.ui.base.BasePresenter;
import com.chooloo.www.callmanager.util.PermissionUtils;

import static com.chooloo.www.callmanager.util.PermissionUtils.RC_DEFAULT;

public class SettingsPresenter<V extends SettingsMvpView> extends BasePresenter<V> implements SettingsMvpPresenter<V> {

    @Override
    public void onRequestPermissionResult(int requestCode, int[] grantResults) {
        if (requestCode == RC_DEFAULT && PermissionUtils.checkPermissionsGranted(grantResults)) {
            mvpView.setupSimSelection();
        }
    }

    @Override
    public void refresh() {
        mvpView.goToMainActivity();
    }

    @Override
    public boolean onListPreferenceChange(Preference preference, Object newValue) {
        mvpView.setListPreferenceSummary(preference, newValue);
        return true;
    }

    @Override
    public boolean onSwitchPreferenceChange(Preference preference, Object newValue) {
        mvpView.setSwitchPreferenceSummary(preference, newValue);
        return true;
    }

    @Override
    public boolean onCheckBoxPreferenceChange(Preference preference, Object newValue) {
        mvpView.setCheckBoxPreferenceSummary(preference, newValue);
        return true;
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/settings_fragment/SettingsPresenter.kt
package com.chooloo.www.koler.ui.settings_fragment

import androidx.preference.Preference
import com.chooloo.www.koler.ui.base.BasePresenter

class SettingsPresenter<V : SettingsMvpView> : BasePresenter<V>(), SettingsMvpPresenter<V> {

    override fun onRequestPermissionResult(requestCode: Int, grantResults: IntArray) {
        mvpView?.setupSimSelection()
    }

    override fun onListPreferenceChange(preference: Preference, newValue: Any): Boolean {
        mvpView?.setListPreferenceSummary(preference, newValue)
        return true
    }

    override fun onSwitchPreferenceChange(preference: Preference, newValue: Any): Boolean {
        mvpView?.setSwitchPreferenceSummary(preference, newValue)
        return true
    }

    override fun onCheckBoxPreferenceChange(preference: Preference, newValue: Any): Boolean {
        mvpView?.setCheckBoxPreferenceSummary(preference, newValue)
        return true
    }

    override fun refresh() {
        mvpView?.goToMainActivity()
    }
}